sims/verilator/Makefile

```diff
@@ -116,48 +179,70 @@ model_header_debug = $(model_dir_debug)/V$(VLOG_MODEL).h
+$(model_mk_debug): $(sim_vsrcs_printf) $(sim_common_files)
 	rm -rf $(model_dir_debug)
 	mkdir -p $(model_dir_debug)
-	$(VERILATOR) $(VERILATOR_OPTS) -o $(sim_debug) --trace -Mdir $(model_dir_debug) -CFLAGS "-include $(model_header_debug)"
+	$(VERILATOR) $(VERILATOR_OPTS) $(EXTRA_SIM_SOURCES) -o $(sim_debug) \
+		$(TRACING_OPTS) \
+		-Mdir $(model_dir_debug) -CFLAGS "-include $(model_header_debug)"
 	touch $@
 
-#########################################################################################
+#============================================================================
 # invoke make to make verilator sim rules
-#########################################################################################
+#============================================================================
 $(sim): $(model_mk) $(dramsim_lib)
 	$(MAKE) VM_PARALLEL_BUILDS=1 -C $(model_dir) -f V$(VLOG_MODEL).mk
 
 $(sim_debug): $(model_mk_debug) $(dramsim_lib)
 	$(MAKE) VM_PARALLEL_BUILDS=1 -C $(model_dir_debug) -f V$(VLOG_MODEL).mk
 
-#########################################################################################
+#============================================================================
 # create a verilator vpd rule
-#########################################################################################
+#============================================================================
 .PRECIOUS: $(output_dir)/%.vpd %.vcd
 $(output_dir)/%.vpd: $(output_dir)/% $(sim_debug)
 	rm -f $@.vcd && mkfifo $@.vcd
 	vcd2vpd $@.vcd $@ > /dev/null &
-	(set -o pipefail && $(sim_debug) $(PERMISSIVE_ON) $(SIM_FLAGS) $(EXTRA_SIM_FLAGS) $(VERBOSE_FLAGS) -v$@.vcd $(PERMISSIVE_OFF) $< </dev/null 2> >(spike-dasm > $<.out) | tee $<.log)
+	(set -o pipefail && $(sim_debug) $(SIM_FLAGS) \
+		$(EXTRA_SIM_FLAGS) $(VERBOSE_FLAGS) \
+		-v$@.vcd $< \
+		</dev/null 2> >(spike-dasm > $<.out) | tee $<.log)
 
 $(output_dir)/none.vpd: $(sim_debug)
 	mkdir -p $(output_dir)
 	rm -f $@.vcd && mkfifo $@.vcd
 	vcd2vpd $@.vcd $@ > /dev/null &
-	(set -o pipefail && $(sim_debug) $(PERMISSIVE_ON) $(SIM_FLAGS) $(EXTRA_SIM_FLAGS) $(VERBOSE_FLAGS) -v$@.vcd $(PERMISSIVE_OFF) none </dev/null 2> >(spike-dasm > $(output_dir)/none.out) | tee $(output_dir)/none.log)
+	(set -o pipefail && $(sim_debug) $(SIM_FLAGS) \
+		$(EXTRA_SIM_FLAGS) $(VERBOSE_FLAGS) -v$@.vcd none \
+		</dev/null 2> >(spike-dasm > $(output_dir)/none.out) \
+		| tee $(output_dir)/none.log)
 
-#########################################################################################
+#============================================================================
 # general cleanup rule
-#########################################################################################
-.PHONY: clean
+#============================================================================
+.PHONY: clean clean-vl8 clean-vl8-debug
 clean:
 	rm -rf $(gen_dir) $(sim_prefix)-*
+
+clean-vl8:
+	rm -rf $(model_dir) $(build_dir)/*.printf.v $(sim)
```

Comment:
Are we opposed to having 1 clean that just clears everything? Also why are you just deleting the print stuff in these targets? IMO a `clean` operation should delete all generated sources, not just one specific source.

Reply:
yes. i found it very useful to blow away the printf's and the stuff that verilator generates while leaving the firrtl output intact

Reply:
for example, if i wanted to enable different printf's, or use a different number of threads

Reply:
I think its ok to have multiple levels of cleaning but they should be named well so you understand whats being removed.

Reply:
@colinschmidt proposals for an alternative to `clean`, `clean-verilator` and `clean-verilator-debug`?

Reply:
I am pro separate cleans. I often just make a change to verilog resource files, then copy them over manually to skip rebuilding chisel. I think `clean`, `clean-verilator`, and `clean-verilator-debug` are fine, but I might suggest `clean`, `clean-sim`, and `clean-sim-debug`, so that we can implement the same feature in VCS, and the targets can have the same name.

Reply:
implemented @jwright6323 's suggestion of `clean-sim`